cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -66,6 +67,7 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	}
 
 	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, queryMetricsFactory)
+	traceReader := factoryadapter.NewTraceReader(spanReader)
```

Comment:
@yurishkuro should we be getting the trace reader via `GetStorageFactoryV2`? I wasn't sure if that was sufficient because we decorate the spanreader above

Reply:
yes, now it should be calling GetStorageFactoryV2. And yes about the decorator. Also, we may want to use otel.MeterProvider in the decorator instead of metrics.Factory (we need to gradually start deprecating metrics.Factory - let's book a ticket for that)

Reply:
@yurishkuro Changed to call `GetStorageFactoryV2`. For the deprecation, this is what `metrics.Factory` currently looks like:
```go
// Factory creates new metrics
type Factory interface {
	Counter(metric Options) Counter
	Timer(metric TimerOptions) Timer
	Gauge(metric Options) Gauge
	Histogram(metric HistogramOptions) Histogram

	// Namespace returns a nested metrics factory.
	Namespace(scope NSOptions) Factory
}
``` 
Looking at OTEL's [meter](https://github.com/open-telemetry/opentelemetry-go/blob/main/metric/meter.go), I see it has a counter, gauge, and histogram. I don't see a Timer - what would be the alternative for that? Also, how do we handle the nesting of namespaces?

Reply:
Are you trying to implement metrics.Factory on top of OTEL MeterProvider? I think it would be great if we could do that, it would minimize our changes across the code base.

Timer doesn't have to be part of the factory API, it can be a util struct done on top of a Histogram - the only difference of timer from Histogram is that its Record() function is strongly typed to Duration.

Name spacing is generally handled by our implementations as a prefix on the metric name, so it would work just the same for OTEL Meter.

Reply:
@yurishkuro Would you able to clarify what you mean by `Are you trying to implement metrics.Factory on top of OTEL MeterProvider?` I was trying to see if we could remove `metrics.Factory` and just replace it with OTEL's Meter Provider. Is there a different approach you were thinking of?

Reply:
I was wrong, we already have `internal/metrics/otelmetrics/factory.go`

Reply:
I guess you were simply asking about v2 version of decorator that wraps Storage V2 API - the decorator can still work off our internal pkg/metrics API.

Reply:
sounds good - so do we need to book any work here?

Reply:
we just need not to drop the metrics for storage - this is related to https://github.com/jaegertracing/jaeger/issues/6219